src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR_OF_DATA = ",";
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_AMOUNT = 1;
+    private static final String SUPPLY_DATA = "supply";
+    private static final String BUY_DATA = "buy";
+    private static final String RESULT_DATA = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeDataToFile(toFileName, getReport(readFromFile(fromFileName)));
+    }
+
+    private String readFromFile(String fileName) {
+        File file = new File(fileName);
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String dataFromFile = bufferedReader.readLine();
+            while (dataFromFile != null) {
+                builder.append(dataFromFile).append(System.lineSeparator());
+                dataFromFile = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File not found" + fileName, e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can not read file" + fileName, e);
+        }
```

Comment:
Don't create two identical `catch` blocks. Either use `multiCatch` ([reference](https://jenkov.com/tutorials/java-exception-handling/catching-multiple-exceptions.html)).

But, here you have a hierarchy. `FileNotFound` is already an `IOException`. So remove the first `catch`.